Library/Formula/ruby.rb

```diff
@@ -106,14 +108,27 @@ def post_install
     config_file.write rubygems_config
 
     # Create the sitedir and vendordir that were skipped during install
-    mkdir_p `#{bin}/ruby -e 'require "rbconfig"; print RbConfig::CONFIG["sitearchdir"]'`
-    mkdir_p `#{bin}/ruby -e 'require "rbconfig"; print RbConfig::CONFIG["vendorarchdir"]'`
+    ruby="#{bin}/ruby#{program_suffix}"
+    %w[sitearchdir vendorarchdir].each do |dir|
+      mkdir_p `#{ruby} -rrbconfig -e 'print RbConfig::CONFIG["#{dir}"]'`
+    end
+
+    # Create the version-specific bindir used by rubygems
+    mkdir_p "#{rubygems_bindir}"
   end
 
   def abi_version
     "2.3.0"
   end
 
+  def program_suffix
+    build.with?("suffix") ? "23" : ""
+  end
```

Comment:
The reason it's like that is because that's a straight copy from the Rubies in `homebrew/versions` and one of my goals was to keep the inter-formula diff as minimal as possible. It's strategically placed next to `abi_version` so it won't be easily overlooked.

I could use `abi_version` to compute the program suffix, if you feel strongly about that.